org.eclipse.wb.swing/src/org/eclipse/wb/internal/swing/model/property/editor/color/CustomColorPickerComposite.java

```diff
@@ -0,0 +1,161 @@
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.Font;
+import org.eclipse.swt.graphics.RGB;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+
+public class CustomColorPickerComposite extends AbstractColorsComposite {
+  private Label colorLabelBlue;
+  private Label colorLabelRed;
+  private Label colorLabelGreen;
+  private Label colorLabelHue;
+  private Label colorLabelSaturation;
+  private Label colorLabelbrightness;
+  private Text colorTextBlue;
+  private Text colorTextRed;
+  private Text colorTextGreen;
+  private Text colorTextHue;
+  private Text colorTextSaturation;
+  private Text colorTextbrightness;
+  private Text previewText;
+  private Composite previewImage;
+  private Text previewTextBackgroundBlack;
+  private Text previewTextBackgroundWhite;
+  ColorSelector colorSelector;
+
+  public CustomColorPickerComposite(Composite parent, int style, AbstractColorDialog colorDialog) {
+    super(parent, style, colorDialog);
+    setLayout(new GridLayout());
+    createContent(this);
+  }
+
+  private void createContent(Composite parent) {
+    Composite color = new Composite(parent, SWT.NONE);
+    GridLayoutFactory.swtDefaults().applyTo(color);
+    GridDataFactory.swtDefaults().applyTo(color);
+    Composite colorSelectorComp = new Composite(color, SWT.NONE);
+    GridLayoutFactory.swtDefaults().applyTo(colorSelectorComp);
+    GridDataFactory.swtDefaults().applyTo(colorSelectorComp);
+    Label buttonLabel = new Label(colorSelectorComp, SWT.NONE);
+    buttonLabel.setText(ModelMessages.CustomColorPicker_btnSelectColor);
+    colorSelector = new ColorSelector(colorSelectorComp);
+    createInfoComposite(color);
+    previewComposite(color);
+    colorSelector.getButton().addSelectionListener(new SelectionListener() {
+      public void widgetSelected(SelectionEvent e) {
+        if (colorSelector.getColorValue() == null) {
+          return;
+        }
+        RGB currentColor = colorSelector.getColorValue();
+        m_colorDialog.setResultColor(
+            new ColorInfo(currentColor.red, currentColor.green, currentColor.blue));
+        colorSelector.setColorValue(currentColor);
+        updatePreview(currentColor);
+        colorTextRed.setText("" + currentColor.red);
+        colorTextGreen.setText("" + currentColor.green);
+        colorTextBlue.setText("" + currentColor.blue);
+        colorTextHue.setText("" + String.format("%.2f", currentColor.getHSB()[0]));
+        colorTextSaturation.setText("" + String.format("%.2f", currentColor.getHSB()[1]));
+        colorTextbrightness.setText("" + String.format("%.2f", currentColor.getHSB()[2]));
+      }
+
+      public void widgetDefaultSelected(SelectionEvent e) {
+        // TODO Auto-generated method stub
+      }
+    });
+  }

```

Comment:
To avoid empty methods, use SelectionAdapter instead of SelectionListener.

you can also use lambda 

    colorSelector.getButton().addListener(SWT.Selection, e->{magic here});